src/hotspot/share/gc/serial/serialHeap.cpp

```diff
@@ -431,7 +455,19 @@ bool SerialHeap::do_young_collection(bool clear_soft_refs) {
     Universe::verify("After GC");
   }
 
-  _young_gen->compute_new_size();
+  if (SharedSerialGCVirtualSpace) {
+    size_t young_gen_size_before = _shared_virtual_space->young_region().byte_size();
+    size_t young_gen_size = _young_gen->compute_new_size();
+
+    bool success = _shared_virtual_space->resize(young_gen_size);
```

Comment:
What should happen if this fails?

Reply:
> What should happen if this fails?

I have added a comment that "execution will continue with the existing heap generation sizes if resizing the VirtualSpace fails"